FreeRTOS-Plus/Demo/AWS/Fleet_Provisioning_Windows_Simulator/Fleet_Provisioning_With_CSR_Demo/DemoTasks/FleetProvisioningDemoExample.c

```diff
@@ -0,0 +1,864 @@
+
+/**
+ * @brief The maximum number of times to run the loop in this demo.
+ *
+ * @note The demo loop is attempted to re-run only if it fails in an iteration.
+ * Once the demo loop succeeds in an iteration, the demo exits successfully.
+ */
+#ifndef fpdemoMAX_DEMO_LOOP_COUNT
+    #define fpdemoMAX_DEMO_LOOP_COUNT    ( 3 )
+#endif
+
+/**
+ * @brief Time in seconds to wait between retries of the demo loop if
+ * demo loop fails.
+ */
+#define fpdemoDELAY_BETWEEN_DEMO_RETRY_ITERATIONS_SECONDS    ( 5 )
+
+/**
+ * @brief Size of buffer in which to hold the certificate signing request (CSR).
+ */
+#define fpdemoCSR_BUFFER_LENGTH                              2048
+
+/**
+ * @brief Size of buffer in which to hold the certificate.
+ */
+#define fpdemoCERT_BUFFER_LENGTH                             2048
+
+/**
+ * @brief Size of buffer in which to hold the certificate id.
+ *
+ * See https://docs.aws.amazon.com/iot/latest/apireference/API_Certificate.html#iot-Type-Certificate-certificateId
+ */
+#define fpdemoCERT_ID_BUFFER_LENGTH                          64
+
+/**
+ * @brief Size of buffer in which to hold the certificate ownership token.
+ */
+#define fpdemoOWNERSHIP_TOKEN_BUFFER_LENGTH                  512
+
+/**
+ * @brief Milliseconds per second.
+ */
+#define fpdemoMILLISECONDS_PER_SECOND                          ( 1000U )
+
+/**
+ * @brief Milliseconds per FreeRTOS tick.
+ */
+#define fpdemoMILLISECONDS_PER_TICK                            ( fpdemoMILLISECONDS_PER_SECOND / configTICK_RATE_HZ )
+
+/**
+ * @brief Status values of the Fleet Provisioning response.
+ */
+typedef enum
+{
+    ResponseNotReceived,
+    ResponseAccepted,
+    ResponseRejected
+} ResponseStatus_t;
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Status reported from the MQTT publish callback.
+ */
+static ResponseStatus_t xResponseStatus;
+
+/**
+ * @brief Buffer to hold the provisioned AWS IoT Thing name.
+ */
+static char pcThingName[ fpdemoMAX_THING_NAME_LENGTH ];
+
+/**
+ * @brief Length of the AWS IoT Thing name.
+ */
+static size_t xThingNameLength;
+
+/**
+ * @brief Buffer to hold responses received from the AWS IoT Fleet Provisioning
+ * APIs. When the MQTT publish callback receives an expected Fleet Provisioning
+ * accepted payload, it copies it into this buffer.
+ */
+static uint8_t pucPayloadBuffer[ democonfigNETWORK_BUFFER_SIZE ];
+
+/**
+ * @brief Length of the payload stored in #pucPayloadBuffer. This is set by the
+ * MQTT publish callback when it copies a received payload into #pucPayloadBuffer.
+ */
+static size_t xPayloadLength;
+
+/**
+ * @brief Global entry time into the application to use as a reference timestamp
+ * in the #prvGetTimeMs function. #prvGetTimeMs will always return the difference
+ * between the current time and the global entry time. This will reduce the chances
+ * of overflow for the 32 bit unsigned integer used for holding the timestamp.
+ */
+static uint32_t ulGlobalEntryTimeMs;
```

Comment:
```suggestion
static uint32_t ulGlobalEntryTimeMs;

```

Reply:
Removed ulGlobalEntryTimeMs in this commit:
https://github.com/FreeRTOS/FreeRTOS/pull/724/commits/0a18eb6432509f18fd6b1015b07ca343ca5d4e32